adam/experiment/live_demo.py

```diff
@@ -0,0 +1,319 @@
+    color_seed: int = 42
+
+    def initialize_curriculum(self) -> None:
+        """Create the curriculum directory"""
+        if self.curriculum_path.is_dir():
+            print(
+                "Curriculum already exists. Configuring situation number automatically."
+            )
+            info_path = self.curriculum_path / "info.yaml"
+            if info_path.exists():
+                with open(info_path, encoding="utf-8") as info_file:
+                    info_yaml = yaml.safe_load(info_file)
+                self.situation_number = int(info_yaml["num_dirs"])
+
+        self.curriculum_path.mkdir(parents=True, exist_ok=True)
+
+    def reset(self, *, initialize_after_reset: bool = True) -> None:
+        """Reset the curriculum directory, optionally reinitialize it."""
+        if self.curriculum_path.is_dir():
+            self.curriculum_path.rmdir()
+
+        if initialize_after_reset:
+            self.initialize_curriculum()
+
+    def preprocess(
+        self, image_path: Path, host: str, stroke_model_path: Path, stroke_python: Path
+    ) -> Path:
+        """Preprocess an image with stroke extraction to configure the curriculum scene."""
+        situation_dir = self.curriculum_path / f"situation_{self.situation_number}"
+        situation_dir.mkdir(parents=True)
+        situation_img_path = situation_dir / "rgb_0.png"
+        segmentation_path = situation_dir / "semantic_0.png"
+        img = cv2.imread(str(image_path))
+        img2 = None
+        try:
+            img2 = cv2.cvtColor(img, cv2.COLOR_RGBA2RGB)
+            print("Image is RGBA")
+        except ValueError:
+            print("Image is RGB")
+            img2 = img
+        finally:
+            cv2.imwrite(str(situation_img_path), img2)
+
+        # Generate a Semantic Segmentation Image
+        with situation_img_path.open("rb") as img_file:
+            im_bytes = img_file.read()
+        im_b64 = base64.b64encode(im_bytes).decode("utf8")
+        headers = {"Content-type": "application/json", "Accept": "text/plain"}
+        payload = json.dumps({"image": im_b64, "other_key": "value"})
+        api = host + "instanceseg"
+        print(f"Processing Image Segmentation from host: {api}")
+        resp = requests.post(api, data=payload, headers=headers)
+
+        data = resp.json()
+
+        print("Generating Segmentation Mask")
+        # 3D array: (N, W, H) where N is number of masks
+        masks = np.asarray(data["masks"], dtype=np.uint8)
+        print(f"Got {len(masks)} masks.")
+
+        # Assert masks don't overlap
+        for i in range(len(masks)):  # pylint: disable=consider-using-enumerate
+            for j in range(i + 1, len(masks)):  # pylint: disable=consider-using-enumerate
+                if np.any(masks[i] & masks[j]):
+                    print(
+                        f"Warning: Masks {i} and {j} overlap. Preferring later mask {j}."
+                    )
+
+        # Compute segmentation IDs
```

Comment:
I want to explain more about the details of how this is computed because it's pretty gnarly, but I can't think of a good comment there, so maybe it is more useful to simply expand on what this expression computes:

```suggestion
        # Compute pixelwise segmentation/mask IDs
        #
        # That is, an input-image-sized array (shape (W, H)) where each entry is either 0 (meaning
        # not in any mask) or a positive integer identifying which of the `N` output masks this pixel
        # belongs to. When a pixel lies in the overlap of several masks, we arbitrarily take the
        # highest-numbered mask.
```